tests/cli/test_main_rename.py

```diff
@@ -4,283 +4,223 @@
+    run_command(Commands.REMOVE, name, "--all")
 
 
 @pytest.fixture
-def env_two():
-    # Setup
-    run(f"conda create -n {TEST_ENV_NAME_2} -y", disallow_stderr=False)
+def name_one(name_rename: str) -> Generator[str, None, None]:
+    """A setup fixture, creates an empty testing environment and removes it at the end if present."""
+    name = uuid.uuid4().hex
+    run_command(Commands.CREATE, name)
 
-    yield
+    yield name
 
-    # Teardown
-    run(f"conda remove --all -y -n {TEST_ENV_NAME_2}", disallow_stderr=False)
+    run_command(Commands.REMOVE, name, "--all")
 
 
 @pytest.fixture
-def env_prefix_one():
-    """Used to get an environment created using -p flag"""
-    # Setup
-    tmpdir = tempfile.mkdtemp()
-    run(f"conda create -p {tmpdir} -y", disallow_stderr=False)
+def name_two() -> Generator[str, None, None]:
+    """A setup fixture, creates an empty testing environment and removes it at the end if present."""
+    name = uuid.uuid4().hex
+    run_command(Commands.CREATE, name)
 
-    yield tmpdir
+    yield name
 
-    # Teardown
-    run(f"conda remove --all -y -p {tmpdir}", disallow_stderr=False)
+    run_command(Commands.REMOVE, name, "--all")
 
 
-def list_envs():
-    out, err, exit_code = run(ENV_LIST_COMMAND)
-    data = json.loads(out)
+@pytest.fixture
+def prefix_rename() -> Generator[Path, None, None]:
+    """A teardown fixture, removes the renamed environment if present."""
+    with tempfile.TemporaryDirectory() as root:
+        prefix = Path(root, uuid.uuid4().hex)
 
-    return (out, err, exit_code), data
+        yield prefix
 
 
-def test_rename_by_name_success(env_one):
-    run(
-        f"conda rename -n {TEST_ENV_NAME_1} {TEST_ENV_NAME_RENAME}",
-        disallow_stderr=False,
-    )
+@pytest.fixture
+def prefix_one(prefix_rename: Path) -> Generator[Path, None, None]:
+    """A setup fixture, creates an empty testing environment and removes it at the end if present."""
+    with tempfile.TemporaryDirectory() as root:
+        prefix = Path(root, uuid.uuid4().hex)
+        run_command(Commands.CREATE, str(prefix))
 
-    assert locate_prefix_by_name(TEST_ENV_NAME_RENAME)
-    with pytest.raises(EnvironmentNameNotFound):
-        locate_prefix_by_name(TEST_ENV_NAME_1)
+        yield prefix
 
-    # Clean up
-    run(
-        f"conda rename -n {TEST_ENV_NAME_RENAME} {TEST_ENV_NAME_1}",
-        disallow_stderr=False,
-    )
 
+@pytest.fixture
+def prefix_two() -> Generator[Path, None, None]:
+    """A setup fixture, creates an empty testing environment and removes it at the end if present."""
+    with tempfile.TemporaryDirectory() as root:
+        prefix = Path(root, uuid.uuid4().hex)
+        run_command(Commands.CREATE, str(prefix))
 
-def test_rename_by_path_success(env_one):
-    with tempfile.TemporaryDirectory() as temp_dir:
-        new_name = str(pathlib.Path(temp_dir).joinpath("new-env"))
-        run(f"conda rename -n {TEST_ENV_NAME_1} {new_name}", disallow_stderr=False)
+        yield prefix
```

Comment:
@jezdez should this `yield` statement also be indented by one? (as suggested [here](https://github.com/conda/conda/pull/12582/files#r1163841617) and [here](https://github.com/conda/conda/pull/12582/files#r1163844302))